client/engine.go

```diff
@@ -275,3 +279,19 @@ func getMsgChan(conn *websocket.Conn) chan commons.Message {
 	}()
 	return messageChan
 }
+
+// handleStatusMsg asynchronously waits for messages from e.StatusChan and
+// displays the message when it arrives.
+func handleStatusMsg() {
+	for msg := range e.StatusChan {
+		e.StatusMsg = msg
+		logger.Infof("got status message: %s", e.StatusMsg)
+
+		e.ShowMsg = true
+		e.Draw()
+		time.Sleep(3 * time.Second)
```

Comment:
What is the effect introduced by `time.Sleep` here, as compared to the previous `time.AfterFunc`?

```go
// Toggle showMsg to false to hide the message.
	_ = time.AfterFunc(5*time.Second, func() {
		e.ShowMsg = false
	})
```

Reply:
`time.AfterFunc` spawns a new goroutine for the function on every call, and the function was modifying the global `editor`. This worked if the `e.SetStatusBar` method was called once at a time, because the separate goroutine could handle the status bar state without blocking. If it was called multiple times (like to display multiple joining messages), some messages were never being shown because `e.ShowMsg` was set to false in one goroutine and never set back to true. Generally, the creation of multiple goroutines all modifying a global variable was hard to reason about.

I think the control flow here better matches the intended behavior of the status bar. Everything happens in one separate goroutine (called with `go handleStatusMsg()`), and within that goroutine everything happens in one thread: listen for a message, display the message for a given amount of time, then stop displaying the message, and repeat the cycle. All status bar modification happens in one thread rather than multiple.